storybook/config/services.yaml

```diff
@@ -6,10 +6,30 @@ services:
         autowire: true
         autoconfigure: true
 
+    PrestaShop\Module\AutoUpgrade\:
+        resource: '../../classes'
+        exclude: '../../classes/**/index.php'
+
+    # makes classes in src/ available to be used as services
+    # this creates a service per class whose id is the fully-qualified class name
     App\:
         resource: '../src/'
         exclude:
             - '../src/DependencyInjection/'
             - '../src/Entity/'
             - '../src/Kernel.php'
 
+    PrestaShop\Module\AutoUpgrade\UpgradeTools\Translator:
+        arguments:
+            $translationsFilesPath: '%kernel.project_dir%/../translations/'
+            $locale: '@=service("request_stack").getCurrentRequest()?.getPayload().all("args")["_locale"]'
```

Comment:
Using this type of expression is often not a good practice, it is better to inject the request stack to do this work in the constructor for several reasons:
- complexity and reduced readability
- performance (they need to be evaluated every time the service is instantiated)
- debugging
- mock in tests

Reply:
I'm not in my confort zone here. Even though I agree with you, I had to define in the service container the parameters of the our module `Translator` class to allow Storybook to boot (issue with the autowiring I guess).

I initially had my own construction of the Translation in the TranslatorBridge, but Symfony did no care about it.

I think sending the whole request is out of the table for our Translator in the module folder.

Reply:
So it may be a better approach to go through a factory https://symfony.com/doc/current/service_container/factories.html

Reply:
Switched on a combo Factory / EventListener